store/mongo/datastore_mongo.go

```diff
@@ -603,6 +603,49 @@ func (db *DataStoreMongo) AddAuthSet(ctx context.Context, set model.AuthSet) err
 	return nil
 }
 
+// UpsertAuthSetStatus inserts a new auth set and if it exists, ensures the
+// AuthSet.Status matches the one in authSet.
+func (db *DataStoreMongo) UpsertAuthSetStatus(ctx context.Context, authSet *model.AuthSet) error {
+	c := db.client.Database(DbName).Collection(DbAuthSetColl)
+
+	if authSet.Id == "" {
+		uid := oid.NewUUIDv4()
+		authSet.Id = uid.String()
+	}
+	id := identity.FromContext(ctx)
+	tenantId := ""
+	if id != nil {
+		tenantId = id.Tenant
+	}
+	authSet.TenantID = tenantId
+	if authSet.Timestamp == nil {
+		ts := time.Now()
+		authSet.Timestamp = &ts
+	}
+
+	fltr := bson.D{
+		{Key: dbFieldTenantID, Value: tenantId},
+		{Key: dbFieldIDDataSha, Value: authSet.IdDataSha256},
+		{Key: dbFieldPubKey, Value: authSet.PubKey},
+	}
+	setOnInsert := *authSet
+	setOnInsert.Status = ""
```

Comment:
why is that empty?

Reply:
I have to clear the status on the `$setOnInsert` since it is set in the `$set` clause. Putting a value in both will lead to a ConflictingValue exception (even if the values are the same). The resulting document that ends up in the DB is always the merged object.